<?xml version="1.0" encoding="UTF-8"?><testrun name="TestGuardedLinkedListDoubleCopy" project="BT-Library-Java" tests="37" started="37" failures="0" errors="26" ignored="0">
  <testsuite name="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.064">
    <testcase name="test_CTOR_emptyList_size0" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0"/>
    <testcase name="test_add_NotEmptyList_size2" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_add_NotEmptyList_size2(TestGuardedLinkedListDoubleCopy.java:60)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_add_NotEmptyList_size3" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_add_NotEmptyList_size3(TestGuardedLinkedListDoubleCopy.java:81)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_add_emptyList_size1" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_add_emptyList_size1(TestGuardedLinkedListDoubleCopy.java:42)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_clear_empty_size0" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0"/>
    <testcase name="test_clear_notEmpty_size0" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_clear_notEmpty_size0(TestGuardedLinkedListDoubleCopy.java:101)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_contains_empty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001"/>
    <testcase name="test_contains_notEmpty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.002">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_contains_notEmpty_false(TestGuardedLinkedListDoubleCopy.java:153)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_contains_notEmpty_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_contains_notEmpty_true(TestGuardedLinkedListDoubleCopy.java:176)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_delete_byIdx_empty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001"/>
    <testcase name="test_delete_byIdx_notEmpty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_delete_byIdx_notEmpty_false(TestGuardedLinkedListDoubleCopy.java:216)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_delete_byIdx_notEmpty_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_delete_byIdx_notEmpty_true(TestGuardedLinkedListDoubleCopy.java:239)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_delete_byVal_empty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0"/>
    <testcase name="test_delete_byVal_notEmpty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_delete_byVal_notEmpty_false(TestGuardedLinkedListDoubleCopy.java:283)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_delete_byVal_notEmpty_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_delete_byVal_notEmpty_true(TestGuardedLinkedListDoubleCopy.java:306)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_get_empty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001"/>
    <testcase name="test_get_notEmpty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_get_notEmpty_false(TestGuardedLinkedListDoubleCopy.java:350)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_get_notEmpty_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_get_notEmpty_true(TestGuardedLinkedListDoubleCopy.java:374)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_indexOf_empty_m1" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.044"/>
    <testcase name="test_indexOf_notEmpty_0" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_indexOf_notEmpty_0(TestGuardedLinkedListDoubleCopy.java:429)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_indexOf_notEmpty_m1" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_indexOf_notEmpty_m1(TestGuardedLinkedListDoubleCopy.java:410)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_insert_on0_empty_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_insert_on0_empty_true(TestGuardedLinkedListDoubleCopy.java:470)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_insert_on0_oneElInTape_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_insert_on0_oneElInTape_true(TestGuardedLinkedListDoubleCopy.java:490)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_insert_on0_twoElInTape_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_insert_on0_twoElInTape_true(TestGuardedLinkedListDoubleCopy.java:511)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_insert_on10_15elOnTape_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_insert_on10_15elOnTape_true(TestGuardedLinkedListDoubleCopy.java:680)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_insert_on15_15elOnTape_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_insert_on15_15elOnTape_true(TestGuardedLinkedListDoubleCopy.java:625)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_insert_on1_empty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0"/>
    <testcase name="test_insert_on1_oneElInTape_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_insert_on1_oneElInTape_true(TestGuardedLinkedListDoubleCopy.java:534)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_insert_on1_twoElInTape_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_insert_on1_twoElInTape_true(TestGuardedLinkedListDoubleCopy.java:557)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_insert_on3_twoElInTape_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_insert_on3_twoElInTape_false(TestGuardedLinkedListDoubleCopy.java:583)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_isEmpty_empty_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0"/>
    <testcase name="test_isEmpty_notEmpty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_isEmpty_notEmpty_false(TestGuardedLinkedListDoubleCopy.java:733)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_set_empty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001"/>
    <testcase name="test_set_notEmpty_false" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_set_notEmpty_false(TestGuardedLinkedListDoubleCopy.java:809)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_set_notEmpty_true" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_set_notEmpty_true(TestGuardedLinkedListDoubleCopy.java:833)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
    <testcase name="test_size_empty_0" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.001"/>
    <testcase name="test_size_notEmpty_2" classname="bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy" time="0.0">
      <error>java.lang.IndexOutOfBoundsException&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.checkOutOfBounds(GuardedLinkedListDoubleCopy.java:98)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.insert(GuardedLinkedListDoubleCopy.java:120)&#13;
	at bt.core.collections.dataTypes.GuardedLinkedListDoubleCopy.add(GuardedLinkedListDoubleCopy.java:22)&#13;
	at bt.tests.core.collections.dataTypes.TestGuardedLinkedListDoubleCopy.test_size_notEmpty_2(TestGuardedLinkedListDoubleCopy.java:768)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)&#13;
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)&#13;
	at java.lang.reflect.Method.invoke(Unknown Source)&#13;
	at junit.framework.TestCase.runTest(TestCase.java:176)&#13;
	at junit.framework.TestCase.runBare(TestCase.java:141)&#13;
	at junit.framework.TestResult$1.protect(TestResult.java:122)&#13;
	at junit.framework.TestResult.runProtected(TestResult.java:142)&#13;
	at junit.framework.TestResult.run(TestResult.java:125)&#13;
	at junit.framework.TestCase.run(TestCase.java:129)&#13;
	at junit.framework.TestSuite.runTest(TestSuite.java:252)&#13;
	at junit.framework.TestSuite.run(TestSuite.java:247)&#13;
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)&#13;
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)&#13;
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)&#13;
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)&#13;
&#13;
</error>
    </testcase>
  </testsuite>
</testrun>
